lib/css_parser/rule_set.rb

```diff
@@ -375,35 +378,38 @@ def create_border_shorthand! # :nodoc:
         end
       end
 
-      @declarations.delete('border-width')
-      @declarations.delete('border-style')
-      @declarations.delete('border-color')
+      BORDER_STYLE_PROPERTIES.each { |prop| @declarations.delete(prop)}
 
       unless values.empty?
         @declarations['border'] = {:value => values.join(' ')}
       end
     end
 
+    DIMENSIONS = [
+      ['margin', %w(margin-top margin-right margin-bottom margin-left)].freeze,
+      ['padding', %w(padding-top padding-right padding-bottom padding-left)],
+      ['border-color', %w(border-top-color border-right-color border-bottom-color border-left-color)],
+      ['border-style', %w(border-top-style border-right-style border-bottom-style border-left-style)],
+      ['border-width', %w(border-top-width border-right-width border-bottom-width border-left-width)],
+    ]
+
     # Looks for long format CSS dimensional properties (margin, padding, border-color, border-style and border-width)
     # and converts them into shorthand CSS properties.
     def create_dimensions_shorthand! # :nodoc:
-      directions = ['top', 'right', 'bottom', 'left']
-
-      {'margin'       => 'margin-%s',
-       'padding'      => 'padding-%s',
-       'border-color' => 'border-%s-color',
-       'border-style' => 'border-%s-style',
-       'border-width' => 'border-%s-width'}.each do |property, expanded|
+      return if @declarations.size < 4
 
-        top, right, bottom, left = ['top', 'right', 'bottom', 'left'].map { |side| expanded % side }
-        foldable = @declarations.select do |dim, val|
-          dim == top or dim == right or dim == bottom or dim == left
-        end
+      DIMENSIONS.each do |property, dimensions|
+        (top, right, bottom, left) = dimensions
         # All four dimensions must be present
-        if foldable.length == 4
+        if dimensions.inject(0) { |c, d| c += 1 if @declarations[d]; c } == 4
```

Comment:
```suggestion
        if dimensions.count { |d| @declarations[d] } == 4
```

Reply:
how are you measuring that ?

```
ruby -e 'GC.disable; s = ObjectSpace.each_object.count; [1,2,3,4].count { |x| x == 1 }; p(ObjectSpace.each_object.count - s)'
3
ruby -e 'GC.disable; s = ObjectSpace.each_object.count; [1,2,3,4].inject(0) { |i, x| i += 1 if  x == 1;  }; p(ObjectSpace.each_object.count - s)'
3
```

Reply:
Good catch. 

Took another look at it. No difference between the two. I was looking at a lot of different variations when working on this and must have mis-remembered that. 😞